src/ingest/src/main/scala/osmesa/ingest/QATiles.scala

```diff
@@ -0,0 +1,124 @@
+package osmesa.ingest
+
+import cats.implicits._
+import com.monovore.decline._
+import geotrellis.proj4.{LatLng, WebMercator}
+import geotrellis.spark.tiling.LayoutDefinition
+import geotrellis.spark.io.kryo.KryoRegistrator
+import geotrellis.spark.io.s3.util.S3RangeReader
+import geotrellis.util.FileRangeReader
+import geotrellis.vector.io.json.{GeoJson, JsonFeatureCollection}
+import org.apache.spark.SparkConf
+import org.apache.spark.serializer.KryoSerializer
+import org.apache.spark.sql.{Column, SparkSession}
+import org.apache.spark.sql.functions
+import org.apache.spark.sql.functions.{array_contains, isnull, not, when}
+import org.locationtech.geomesa.spark.jts._
+import org.locationtech.jts.geom._
+import vectorpipe._
+import vectorpipe.functions.osm._
+import vectorpipe.vectortile._
+
+import java.net.URI
+
+case class QATilesPipeline(geometryColumn: String, baseOutputURI: URI) extends Pipeline {
+  val layerMultiplicity = LayerNamesInColumn("layers")
+
+  override def simplify(geom: Geometry, layout: LayoutDefinition): Geometry = Simplify.withJTS(geom, layout)
+
+  override def clip(geom: Geometry, key: geotrellis.spark.SpatialKey, layoutLevel: geotrellis.spark.tiling.LayoutLevel): Geometry =
+    Clipping.byLayoutCell(geom, key, layoutLevel)
+}
+
+object QATiles extends CommandApp(
+  name="qa-tile-generator",
+  header="Create QA tiles from OSM ORC file",
+  main = {
+    /* CLI option handling */
+    val minZoomOpt = Opts.option[Int]("minZoom", help = "Smallest (least resolute) zoom level to generate").withDefault(0)
+    val maxZoomOpt = Opts.option[Int]("maxZoom", help = "Largest (most resolute) zoom level to generate").withDefault(15)
+    val aoiOpt = Opts.option[URI]("aois", help = "Location of GeoJSON file giving areas of interest").withDefault(new URI(""))
+      .validate("AOI file must be a .geojson file") { uri => uri.toString.isEmpty || uri.toString.endsWith(".geojson") }
+    val orcArg = Opts
+      .argument[URI]("source ORC file")//, help = "Location of the ORC file containing geometries to process")
+      .validate("URI to ORC must have an s3 or file scheme") { _.getScheme != null }
+      .validate("orc must be an S3 or file Uri") { uri =>
+        uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+      }
+      .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
```

Comment:
When writing outputs with multiple partitions, it puts individual ORC files into a directory. Unless we want to enforce that the containing directory include `.orc`, let's drop this.